spec/System/HapistranoConfigSpec.hs

```diff
@@ -36,6 +39,27 @@ spec =
         Yaml.loadYamlSettings ["fixtures/git_repository_config.yaml"] [] Yaml.useEnv
           >>= (`shouldBe` defaultConfiguration)
 
+    describe "Config From JSON instances" $ do
+      context "when contains unique targets" $
+        it "parses all unique targets" $
+          let mbHosts = fmap targetHost <$> configHosts <$> A.decode content
+              content =
+                "{\"targets\":[{\"host\":\"user@app1.com\"},{\"host\":\"user@app2.com\"}],\"deploy_path\":\"/tmp\",\"local_directory\":\"/\"}"
+          in maybe
```

Comment:
@CristhianMotoche try comparing the values directly without using a `maybe` function:

```hs
let mHosts = fmap targetHost <$> configHosts <$> A.eitherDecode' content
    content = ...
in mHosts == Right ["user@app1.com", "user@app2.com"]
```

Also, I would recommend using `eitherDecode'` instead of `decode` to retrieve the original error message

Reply:
Makes sense @sestrella Thank!

Reply:
Thanks for the suggestion @sestrella Please, let me know if it looks better now. 👍